src/main/java/core/basesyntax/RectangleSupplier.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class RectangleSupplier {
+    private Random random = new Random();
+    private RandomSize size = new RandomSize();
+    private ColorSupplier color = new ColorSupplier();
+
+    public Figure getRandomRectangle() {

```

Comment:
Consider which variables should be local in the method and which should be class-level. It seems like the `size` and `color` instances could be created inside the `getRandomRectangle` method if they are not used elsewhere in the class.